test/scenario/multi-backend/fabfile/validate.py

```diff
@@ -0,0 +1,188 @@
+import hashlib
+import json
+import io
+import os
+
+import swiftclient.client
+
+from fabric.api import task
+from swiftclient.exceptions import ClientException
+
+AUTH_ENDPOINT = 'http://{host:s}:8080/auth/v1.0'
+
+
+def hash_dataset(directory):
+    """
+    SHA1 hash the content of each file in the given directory.
+
+    :param directory: containing directory of files to compute hashes of
+    :type directory: string
+    :return: dictionary mapping filename to a hash of its content
+    """
+    file_hashes = {}
+    for filename in os.listdir(directory):
+        path = os.path.join(directory, filename)
+        if os.path.isfile(path):
+            with io.open(path, 'rb') as f:
```

Comment:
I tried to upload the /etc directory and this failed with:

```
with io.open(path, 'rb') as f:
  IOError: [Errno 13] Permission denied: '/etc/at.deny'
```
I suggest you add something like
```
if not os.access(path, os.R_OK):
  print("Skipping upload of file {0:s}: file is read protected").format(path)
  continue
```
Or use a try/except block.

I know it seems overkill but if CSE are going to run this, let's be super safe :)